nat-lab/tests/test_client_basic_stun.py

```diff
@@ -24,3 +29,71 @@ async def test_client_basic_stun(connection_tag: ConnectionTag, public_ip: str)
     async with new_connection_by_tag(connection_tag) as connection:
         ip = await testing.wait_long(stun.get(connection, config.STUN_SERVER))
         assert ip == public_ip, f"wrong public ip for the client {ip}"
+
+
+@pytest.mark.asyncio
+# Regression test for LLT-4306
+async def test_infinite_loop_in_stun_when_nonet() -> None:
```

Comment:
Two things here:
 - I'm not sure if this is the right file to place it (it looks a bit obsolete TBH) - I would either move it to a separate file, or just to direct connections,
 - What do you think about replacing `_when_nonet` with something more verbose? Or just getting rid of it?